core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala

```diff
@@ -287,9 +287,10 @@ object Boot extends IOApp with LazyLogging {
       val servicePerimeterService = new ServicePerimeterService(slickDataSource, gcsDAO, servicePerimeterConfig)
 
       val multiCloudWorkspaceConfig = MultiCloudWorkspaceConfig.apply(conf)
+      val billingProfileManagerClientProvider = new HttpBillingProfileManagerClientProvider(conf.getStringOption("billingProfileManager.baseUrl"))
       val billingProfileManagerDAO = new BillingProfileManagerDAOImpl(
         samDAO,
-        new HttpBillingProfileManagerClientProvider(conf.getStringOption("billingProfileManager.baseUrl")),
+        billingProfileManagerClientProvider,
```

Comment:
I wasn't sure if we want to pass billingProfileManagerDAO instead of just the client. I need the client to be able to look up the profile during workspace creation.

Reply:
We should pass the DAO rather than the raw client and add a `getProfile` method, to avoid breaking the abstraction.  That is, BillingProfileManagerDAO should really be the only part of our code that knows about the HTTP client providers.

Reply:
That is what I suspected. I have pushed up the change, with a question.